lua/packer.lua

```diff
@@ -167,6 +168,108 @@ packer.use_rocks = function(rock)
   end
 end
 
+-- Merges new into old; returns merged spec.
+local function merge_specs(old, new)
+  local name = new.short_name
+
+  -- disable: Warn & ignore on disabling a previously enabled plugin
+  if old.disable then
+    if not new.disable then log.warn('Trying to disable previously non-disabled mod "' .. name .. '", ignoring') end
+  else
+    old.disable = new.disable
+  end
+
+  -- branch, tag, commit, lock: Must all be the same, else warning
+  for _, key in ipairs{'branch', 'commit', 'lock', 'tag'} do
+    if new[key] ~= old[key] then
+      log.warn('Plugin "' .. name .. '" wants to overwrite "' .. key .. '" setting. Keeping original value')
+    end
+  end
+
+  -- opt
+  -- - Conditions and opt_default have already been handled in manage()
+  -- - Rule of thumb: true < nil < false. Set the lowest necessary--if nil suffices, don't set false
+  -- - For manual_opt:
+  --   - If result is start, remove mo. Else, if omo or nmo, keep mo.
+  --   - There is no situation where opt = falsy but manual_opt = true
+  --
+  -- old new result (1 = true, - = nil, 0 = false, s = start, o = opt)
+  -- ---+---+------
+  --   -   -     -s
+  --   1   -     -s
+  --   -   1     -s
+  --   0   0     0s
+  --   1   0     0s
+  --   0   1     0s
```

Comment:
Same question here - what if `new.opt = true` because it's adding some lazy-loading?

Reply:
Oof, I totally missed that, thank you. I completely forgot that a lot of keys implicitly enable `opt`.